docs(hip): Add link to Cooperative Groups double-buffered tile example (#8846)
ROCm/rocm-systems: projects/hip/docs/tutorial/cooperative_groups_tutorial.rst

==========

With cooperative groups, you can easily use custom partitions to create custom tiles for custom solutions. You can find the complete code at `cooperative groups ROCm example. <https://github.com/ROCm/rocm-examples/tree/develop/HIP-Basic/cooperative_groups>`_

Additional cooperative groups examples demonstrate more recent features of the API:

> - `double-buffered tile <https://github.com/ROCm/rocm-examples/tree/develop/HIP-Basic/cooperative_groups_double_buffered_tile>`_ combines ``memcpy_async`` with a split barrier.

Reviewer: Is the link correct? It's leading to 404 error.

<img width="1375" height="343" alt="Image" src="https://github.com/user-attachments/assets/a9b27df0-4903-4f46-b701-c4851cf96393" />

Author: Yes, it *will* live there once the corresponding PR on rocm-examples is merged. https://github.com/ROCm/rocm-examples/pull/486

==========

> With cooperative groups, you can easily use custom partitions to create custom tiles for custom solutions. You can find the complete code at `cooperative groups ROCm example. <https://github.com/ROCm/rocm-examples/tree/develop/HIP-Basic/cooperative_groups>`_

Reviewer 2: ```suggestion
With cooperative groups, you can use custom partitions to create custom tiles for custom solutions. For complete code, see `cooperative groups ROCm example. <https://github.com/ROCm/rocm-examples/tree/develop/HIP-Basic/cooperative_groups>`_
```

> Additional cooperative groups examples demonstrate more recent features of the API:

Reviewer 2: ```suggestion
The following additional cooperative group examples demonstrate the latest API features:
```
